filament/src/Camera.cpp

```diff
@@ -234,6 +235,34 @@ Frustum FCamera::getFrustum(mat4 const& projection, mat4f const& viewMatrix) noe
     return Frustum(mat4f{ projection * viewMatrix });
 }
 
+// ------------------------------------------------------------------------------------------------
+
+CameraInfo::CameraInfo(FCamera const& camera) noexcept {
+    projection         = mat4f{ camera.getProjectionMatrix() };
+    cullingProjection  = mat4f{ camera.getCullingProjectionMatrix() };
+    model              = camera.getModelMatrix();
+    view               = camera.getViewMatrix();
+    zn                 = camera.getNear();
+    zf                 = camera.getCullingFar();
+    ev100              = Exposure::ev100(camera);
+    f                  = (FCamera::SENSOR_SIZE * (float)projection[1][1]) * 0.5f;
+    A                  = f / camera.getAperture();
+}
+
+CameraInfo::CameraInfo(FCamera const& camera, const math::mat4f& worldOriginCamera) noexcept {
+    const mat4f modelMatrix{ worldOriginCamera * camera.getModelMatrix() };
+    projection         = mat4f{ camera.getProjectionMatrix() };
+    cullingProjection  = mat4f{ camera.getCullingProjectionMatrix() };
+    model              = modelMatrix;
+    view               = FCamera::getViewMatrix(model);
+    zn                 = camera.getNear();
+    zf                 = camera.getCullingFar();
+    ev100              = Exposure::ev100(camera);
+    f                  = (FCamera::SENSOR_SIZE * (float)projection[1][1]) * 0.5f;
```

Comment:
Should we add a `getFocalLength()` to the public `Camera` API?

Reply:
Yes. It would make sense. I'll do that in a separate PR.